tests/test_annotated.py

```diff
@@ -233,6 +233,7 @@ class _(BaseModel):
         x: Annotated[CustomType, GroupedMetadataMarker(), PydanticMetadata()]
 
     assert calls == [
+        'GroupedMetadataMarker:iter',
```

Comment:
It seems like we end up calling `GroupedMetadataMarker` once more than we did before, I'm not sure why. But `GroupedMetadata` should _definitely_ be idempotent, so it shouldn't really be a problem.